Change CONTRIBUTING docs for atest new test framework (#722)
MarketSquare/robotframework-robocop: CONTRIBUTING.rst

Acceptance tests
''''''''''''''''

> Acceptance tests tests if Robocop rules reports issues in test data files.

Reviewer: ```suggestion
Acceptance tests check if Robocop rules report issues in test data files.
```

> They are located in ``tests/atest/rules/{rules_category}`` directories.
> Each rule have their subdirectory with the name of the rule. Hyphens in the
> name are replaced by underscored. For example ``rule-name`` from ``comments``
> category rule should have ``tests/atest/rules/comments/rule_name`` directory.
> Inside each directory there should be empty ``__init__.py`` file, ``test_rule.py``
> file containing pytest tests and test data and expected data used by tests.

Reviewer: ```suggestion
They are located in ``tests/atest/rules/{rules_category}`` directories.
Each rule has its subdirectory with the name of the rule. Hyphens in the
name are replaced by underscores. For example, ``rule-name`` from ``comments``
category rule should have ``tests/atest/rules/comments/rule_name`` directory.
Inside each directory there should be an empty ``__init__.py`` file, ``test_rule.py``
file containing pytest tests, test data and expected data used by the tests.
```

Acceptance tests should use ``tests.atest.utils.RuleAcceptance`` class that

> contains helper methods and assertions for tests purpose.

Reviewer: ```suggestion
contains helper methods and assertions for the tests purpose.
```

> Example of simple test::

Reviewer: ```suggestion
Example of a simple test::
```

Example of the expected file::

    test.robot:8:1 [E] 0803 Multiple variables with name '${V AR}' in Variables section (first occurrence in line 6). Note that Robot Framework is case-insensitive

Issues are reported using following format: ``{source}:{line}:{col} [{severity}] {rule_id} {desc}``.
If your test data file is inside subdirectory, the path to file should use ``${/}`` as path separator::

    suite_dir{/}__init__.robot:4:1 [W] 0806 Duplicated metadata 'some text' (first occurrence in line 2)

If the rule behaves differently depending on the Robot Framework version, or it is enabled only for
specific version, it is possible to set target version of the tests using version specifiers::

    from tests.atest.utils import RuleAcceptance

    class TestRule(RuleAcceptance):
        def test_rule(self):
            self.check_rule(expected_file="expected_output.txt", target_version=">=5.0")

        def test_rule_rf3(self):
            self.check_rule(expected_file="expected_output_rf4.txt", target_version="==4.1.3")

        def test_rule_rf4(self):
            self.check_rule(expected_file="expected_output_rf3.txt", target_version="==3.2.2")

> You can provide rule custom configuration using ``config`` argument. It accepts either string or list::

Reviewer: ```suggestion
You can provide custom configuration for the rule using ``config`` argument. It accepts either string or list::
```
